### PR TITLE
the groovy file should wait for simpleci.conf file to generate before…

### PR DESCRIPTION
**Pull request by nbansal16**
… execution. Logic is added for that.

### CODE REVIEW

File: init.groovy.d/bitbucket-config.groovy [hunk 1]
```diff
@@ -58,39 +64,55 @@ import java.util.Base64;
...
-			String id = java.util.UUID.randomUUID().toString()
-			// Decode password using base64
-			byte[] valueDecoded = Base64.getDecoder().decode(serverConfig.password);
-			Credentials c = (Credentials) new UsernamePasswordCredentialsImpl(CredentialsScope.GLOBAL,id, serverConfig.username, serverConfig.username, new String(valueDecoded))
-			SystemCredentialsProvider.getInstance().getStore().addCredentials(Domain.global(), c)
+		def file 					= new File("$home_dir/simpleci.conf")
+		int i=1;
+		// check if the file exists at the location. if not wait for it for 2 minutes. Repeat this process for 5 times.	
+		while(!file.exists() && i<6){
+			out.println ("I am a test info log")
```
Comment (amit04pandey) [inline init.groovy.d/bitbucket-config.groovy]: out.println ("Waiting for SimpleCI.conf file : Iteration $i")

File: init.groovy.d/bitbucket-config.groovy [hunk 2]
```diff
@@ -58,39 +64,55 @@ import java.util.Base64;
...
+			i++;
+			Thread.sleep(120000)
+		}
+		//if the file not read even after 10 minutes, print the error message else read the file and continue.
+		if(i==6){
+			out.println ("Config file not present at the location. Please generate the file and try again later")
```
Comment (amit04pandey) [inline init.groovy.d/bitbucket-config.groovy]: out.println ("Config file not present at the location $home_dir/simpleci.conf. Please generate the file and try again later")

File: init.groovy.d/config-init.groovy [hunk 3]
```diff
@@ -19,75 +19,97 @@ import java.util.Base64;
...
+	// Get the out variable
+	def config = new HashMap()
+	def bindings = getBinding()
+	config.putAll(bindings.getVariables())
+	def out = config['out']
+	int i=1;
 	
+	// check if the file exists at the location. if not wait for it for 2 minutes. Repear this process for 5 times.	
+	while(!file.exists() && i<6){
+		out.println ("I am a test info log")
```
Comment (amit04pandey) [inline init.groovy.d/config-init.groovy]: similar comment as bitbucket-init.groovy

File: init.groovy.d/config-init.groovy [hunk 4]
```diff
@@ -19,75 +19,97 @@ import java.util.Base64;
...
+		i++;
+		Thread.sleep(120000)
+	}
+	//if the file not read even after 10 minutes, print the error message else read the file and continue.
+	if(i==6){
+		out.println ("Config file not present at the location. Please generate the file and try again later")
```
Comment (amit04pandey) [inline init.groovy.d/config-init.groovy]: similar comment as bitbucket-init.groovy to specify the target location in error message

**Review by amit04pandey — CHANGES_REQUESTED** [review]
Please review the comments and make corresponding changes

**Review by Goldgod — APPROVED** [review]
only return statement having doubt, instead we could try with sys.exit. Let me merge and test it out once.

**Pull request merged**